### PR TITLE
Rewrite of backend

### PR DESCRIPTION
**Pull request by katelynsills**
## Description
Major rewrite of backend. I'll be splitting this up into PRs that are easier to review, but this is a lot cleaner - all tests pass!

## Test Plan
`yarn test`

### CODE REVIEW

**Comment by karenmrose** [comment]
@katelynsills Can you merge the latest `master` into this PR? Then it should deploy to a review app so we can test and make sure the refactor didn't break any functionality.

File: server/data-import/old/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Import Data
```
Comment (karenmrose) [inline server/data-import/old/README.md]: @katelynsills Are these files in the "old" directory used anymore?

File: server/locations/schema/mongoose-schema.js [hunk 2]
```diff
@@ -25,17 +25,16 @@ const userDateSchema = new mongoose.Schema({
...
+	if (_.isUndefined(this.address)) return ''
+	const getProperty = property => _.get(this.address, property, '')
+
+	return [
+		getProperty('line1'),
+		getProperty('line2'),
+		getProperty('city'),
+		getProperty('state'),
+		getProperty('zip'),
+	].join(' ')
```
Comment (karenmrose) [inline server/locations/schema/mongoose-schema.js]: Nice, so much cleaner!

File: server/routes/api/pregnancy-centers.js [hunk 3]
```diff
@@ -0,0 +1,253 @@
+const router = require('express').Router()
```
Comment (karenmrose) [inline server/routes/api/pregnancy-centers.js]: Omg yay, love that each "domain" is in its own file! 👏

**Comment by karenmrose** [comment]
This is so awesome! It's so much cleaner and easier to read. Thanks for taking this on!

I think the only thing that is a little weird is the "old" directory...if we're still using it, it could be confusing to other devs. Ideally I think we would either delete the entire directory (we could always go back and find it using git if we need it) or remove the "old" from the path. Is there a reason you'd like to keep it though?

**Comment by katelynsills** [comment]
> This is so awesome! It's so much cleaner and easier to read. Thanks for taking this on!
> 
> I think the only thing that is a little weird is the "old" directory...if we're still using it, it could be confusing to other devs. Ideally I think we would either delete the entire directory (we could always go back and find it using git if we need it) or remove the "old" from the path. Is there a reason you'd like to keep it though?

I'm ok getting rid of the `old` directory!

**Pull request merged**